lib/storage/data/LocationConstraintParser.js

```diff
@@ -76,19 +77,26 @@ function parseLC(config, vault) {
                 // https or http proxy
                 connectionAgent = new HttpsProxyAgent(options);
             } else {
-                connectionAgent = sslEnabled ?
-                    new https.Agent(httpAgentConfig, { maxSockets: false }) :
-                    new http.Agent(httpAgentConfig, { maxSockets: false });
+                const agentOptions = { ...httpAgentConfig, maxSockets: false };
+
+                connectionAgent = sslEnabled
+                    ? new https.Agent(agentOptions)
+                    : new http.Agent(agentOptions);
             }
             const httpOptions = { agent: connectionAgent, timeout: 0 };
             const s3Params = {
                 endpoint: `${protocol}://${endpoint}`,
                 region,
+                requestHandler: new NodeHttpHandler({
+                    socketTimeout: 0,
```

Comment:
From this doc it seems we should use `requestTimeout`

https://github.com/aws/aws-sdk-js-v3/blob/main/supplemental-docs/CLIENTS.md#request-handler-requesthandler